warehouse/api.go

```diff
@@ -87,27 +89,50 @@ type UploadAPIT struct {
 	dbHandle          *sql.DB
 	log               logger.LoggerI
 	connectionManager *controlplane.ConnectionManager
-	isHosted          bool
+	isMultiWorkspace  bool
 }
 
 var UploadAPI UploadAPIT
 
-func InitWarehouseAPI(dbHandle *sql.DB, log logger.LoggerI) {
-	workspaceToken := config.GetWorkspaceToken()
-	isMultiWorkspace := config.GetEnvAsBool("HOSTED_SERVICE", false)
-	if isMultiWorkspace {
-		workspaceToken = config.GetEnv("HOSTED_SERVICE_SECRET", "password")
+func InitWarehouseAPI(dbHandle *sql.DB, log logger.LoggerI) error {
+	deploymentType, err := deployment.GetFromEnv()
+	if err != nil {
+		pkgLogger.Errorf("error getting deployment type: %s", err.Error())
+		return err
+	}
+	var connectionIdentifier string
```

Comment:
Can we move this to a separate function?

Reply:
+1